src/icrc1_auction_api_new.mo

```diff
@@ -128,12 +128,125 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
     };
   };
 
-  var auction : ?Auction.Auction = null;
-  var assets : Vec.Vector<AssetInfo> = Vec.new();
+  let quoteAssetId : Auction.AssetId = 0;
+
+  let assets : Vec.Vector<AssetInfo> = Vec.map<StableAssetInfo, AssetInfo>(
+    assetsDataV3,
+    func(x) {
```

Comment:
For readability I think I would define this function beforehand, and give it a name, instead of defining it here inline.

Also, can we re-use this function? There is some of the same code repeated in `registerAsset_`. Can we use it there as well?

Reply:
Done